nil/internal/network/host.go

```diff
@@ -15,46 +17,52 @@ import (
 	"github.com/libp2p/go-libp2p/p2p/security/noise"
 	quic "github.com/libp2p/go-libp2p/p2p/transport/quic"
 	"github.com/libp2p/go-libp2p/p2p/transport/tcp"
+	"github.com/rs/zerolog"
 )
 
 type Host = host.Host
 
 var defaultGracePeriod = connmgr.WithGracePeriod(time.Minute)
 
-func getCommonOptions(ctx context.Context, privateKey libp2pcrypto.PrivKey) ([]libp2p.Option, error) {
-	cm, err := connmgr.NewConnManager(100, 400, defaultGracePeriod)
+func getCommonOptions(ctx context.Context, conf *Config) ([]libp2p.Option, zerolog.Logger, error) {
+	pid, err := peer.IDFromPublicKey(conf.PrivateKey.GetPublic())
 	if err != nil {
-		return nil, err
+		return nil, zerolog.Nop(), err
 	}
 
-	pid, err := peer.IDFromPublicKey(privateKey.GetPublic())
+	logger := internal.Logger.With().
+		Stringer(logging.FieldP2PIdentity, pid).
+		Logger()
+
+	cm, err := connection_manager.NewConnectionManagerWithPeerReputationTracking(
+		ctx, conf.ConnectionManagerConfig, logger, 100, 400, defaultGracePeriod)
 	if err != nil {
-		return nil, err
+		return nil, logger, err
 	}
 
 	metrics, err := internal.NewMetricsReporter(ctx, pid)
 	if err != nil {
-		return nil, err
+		return nil, logger, err
 	}
 
 	return []libp2p.Option{
 		libp2p.Security(noise.ID, noise.New),
 		libp2p.ConnectionManager(cm),
-		libp2p.Identity(privateKey),
+		libp2p.Identity(conf.PrivateKey),
 		libp2p.BandwidthReporter(metrics),
-	}, nil
+	}, logger, nil
 }
 
 // newHost creates a new libp2p host. It must be closed after use.
-func newHost(ctx context.Context, conf *Config) (Host, error) {
+func newHost(ctx context.Context, conf *Config) (Host, zerolog.Logger, error) {
 	addr := conf.IPV4Address
 	if addr == "" {
 		addr = "0.0.0.0"
 	}
 
-	options, err := getCommonOptions(ctx, conf.PrivateKey)
+	options, logger, err := getCommonOptions(ctx, conf)
```

Comment:
Maybe we need to call `getCommonOptions` outside `newHost`. I don't like that newHost returns logger now.

Reply:
I don't have a good opinion on that. I can do that if you think it's best.

Reply:
Frankly, this would affect `newClient`/`NewClientManager`, and thus the private key work that happens there. I'm not sure how good and timely this is. If not critical, we can make some changes here as a separate PR.